lib/collection.js

```diff
@@ -0,0 +1,290 @@
+        this.set(data);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    isChanged: function (branch) {
+        return this._models.some(function (model) {
+            return model.isChanged(branch);
+        });
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    commit: function (branch) {
+        return this._models.reduce(function (changed,  model) {
+            return model.commit(branch) || changed;
+        }, false);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Collection}
+     */
+    revert: function (branch) {
+        this._models.forEach(function (model) {
+            model.revert(branch);
+        });
+        return this;
+    },
+
+    /**
+     * @returns {Array.<Object>}
+     */
+    toJSON: function () {
+        return this._models.map(function (model) {
+            return model.toJSON();
+        });
+    },
+
+    /**
+     * @param {Number} index
+     * @returns {Model}
+     */
+    at: function (index) {
+        return this._models[index];
+    },
+
+    /**
+     * @param {*} id
+     * @returns {?Model}
+     */
+    get: function (id) {
+        return this._modelsIdsMap[id] || null;
+    },
+
+    /**
+     * @param {Function} fn
+     * @param {Object} [ctx]
+     * @returns {?Model}
+     */
+    find: function (fn, ctx) {
+        var found = null;
+        this.some(function (model, index, array) {
+            var isFound;
+            if (ctx) {
+                isFound = fn.call(ctx, model, index, array);
+            } else {
+                isFound = fn(model, index, array);
+            }
+            if (isFound) {
+                found = model;
+                return true;
+            }
+        });
+        return found;
+    },
+
+    /**
+     * @param {Object} conditions
+     * @returns {Array.<Model>}
+     */
+    where: function (conditions) {
+        return this.filter(this._isMatchConditions.bind(this, conditions));
+    },
+
+    /**
+     * @param {Object} conditions
+     * @returns {?Model}
+     */
+    findWhere: function (conditions) {
+        return this.find(this._isMatchConditions.bind(this, conditions));
+    },
+
+    /**
+     * @param {String} attr
+     * @returns {Array}
+     */
+    pluck: function (attr) {
+        return this.map(function (model) {
+            return model.get(attr);
+        });
+    },
+
+    /**
+     * @param {(Model|Object|Array.<(Model|Object)>)} models
+     * @param {Object} [options]
+     * @param {Number} [options.at]
+     * @returns {Collection}
+     */
+    add: function (models, options) {
+        var at = options && options.at;
+        models = Array.isArray(models) ? models : [models];
+        models.forEach(function (model, index) {
+            var currentOptions;
+            model = this._prepareModel(model);
+            if (!this._isExists(model)) {
```

Comment:
Ok. let it be as is, if we'll find case, when it will be required, we can change something. But I think that uniqueness enforcement is not not expected from the begining. At least, if we implement this we should write in documentation, about this.